### PR TITLE
Fixes #8488.

### LINKED ISSUE #8488: Add test case for DatabaseBackendHandler impl
- [ ] Add test case for SchemaAssignedDatabaseBackendHandler
- [ ] Add test case for UnicastDatabaseBackendHandler
- [ ] Add test case for DatabaseBackendHandlerFactory

### PR DESCRIPTION
**Pull request by arthasking123**
Fixes #8488.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/impl/UnicastDatabaseBackendHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UnicastDatabaseBackendHandlerTest {
```
Comment (tristaZero) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/impl/UnicastDatabaseBackendHandlerTest.java:58]: `final` class is suggested.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/impl/UnicastDatabaseBackendHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    @Mock
+    private DatabaseCommunicationEngineFactory databaseCommunicationEngineFactory;
+
+    @Mock
+    private DatabaseCommunicationEngine databaseCommunicationEngine;
+
+    @Before
+    public void setUp() throws IllegalAccessException, NoSuchFieldException, SQLException {
+
```
Comment (tristaZero) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/impl/UnicastDatabaseBackendHandlerTest.java:77]: Please remove this reductant blank line.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/impl/SchemaAssignedDatabaseBackendHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    @Mock
+    private DatabaseCommunicationEngineFactory databaseCommunicationEngineFactory;
+
+    @Mock
+    private DatabaseCommunicationEngine databaseCommunicationEngine;
+
+    @Before
+    public void setUp() throws IllegalAccessException, NoSuchFieldException, SQLException {
+
```
Comment (tristaZero) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/impl/SchemaAssignedDatabaseBackendHandlerTest.java:76]: Please remove this reductant blank line.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/impl/SchemaAssignedDatabaseBackendHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SchemaAssignedDatabaseBackendHandlerTest {
```
Comment (tristaZero) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/impl/SchemaAssignedDatabaseBackendHandlerTest.java:57]: IMO, `final` is preferable.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/DatabaseBackendHandlerFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import org.apache.shardingsphere.sql.parser.sql.common.statement.SQLStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dal.DALStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dal.SetStatement;
+import org.junit.Test;
+import org.mockito.Mock;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class DatabaseBackendHandlerFactoryTest {
```
Comment (tristaZero) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/data/DatabaseBackendHandlerFactoryTest.java:33]: IMO, `final` is preferable.

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi @arthasking123 Thanks for your PR. :)
It looks great practically, except for some small items to improve.

**Comment by arthasking123** [comment]
modified

**Review by tristaZero — APPROVED** [review]
@arthasking123 Spotless!

**Pull request merged**